assets/blocks/course-outline/module-block/edit.js

```diff
@@ -49,7 +49,7 @@ export const EditModuleBlock = ( props ) => {
 
 	// Get the border setting from the parent if none is set.
 	useEffect( () => {
-		if ( undefined === bordered ) {
+		if ( undefined === bordered && undefined !== moduleBorder ) {
```

Comment:
Picking up the parent border option does not seem to work here — since this attribute is always either `true` or `false`, and starts out as `true` now. Setting the default value for the `bordered` attribute of the module block to `null`, and checking for `null` in this check does work, but also breaks editor **Undo**. 

Generally, calling `setAttributes` (or `updateBlockAttributes` on the store) should probably be avoided in effects, and mostly only used when a user changes something. 

Inheriting the parent style for new blocks looks like a bigger task. Attributes need to have a neutral default `unset` value, and their controls in block settings also need a neutral default state. And similar to the `withDefaultStyle` HOC, it should probably be done by appending the parent style to the class list in both the editor and frontend — or just with CSS.

Reply:
Argh! I missed that `setAttributes` modifies the edit history again 😕 I remember testing undo functionality when applying the parent border to children but this initialization was probably added after that.

Inheriting the parent style is harder since there are many edge cases coming from the preview blocks in the 'Styles' selector. Fortunately the border is a custom setting now and not a style so we can handle it more easily. We could also probably get rid of `applyBorder` from course block if are able to go with a CSS solution.

Reply:
TIL about the "undo" issues with @yscik in another PR. 😁

> CSS solution

Do you mean about that solution to just add a class in the outline to apply the border in the modules (something like `.outline.with-module-border .module { border: ... }`)? And override it in the modules (when it's set in the module level).

If yes, I support this idea because I think it can be a very clean solution. But IIRC, we had some issue that we didn't want this behavior, right @gkaragia?

Reply:
The issue would apply more to 'Styles' selector. So we have two options on how this could work.

1. If the users sets a value in the module level, it will always have higher priority of what we set in the outline level.
2. When the user does a change in either levels, the change is applied. It is applied to all modules if it is done in the outline level otherwise it is applied to the selected module.

In general I thought that the second one is preferable since the first one has some cases where the user is doing a change and nothing is updated in the frontend. The second also can't be done with CSS since if both levels have a value set, CSS will always favor one of the two.

For the first case, as Peter said, we will need to have three values for the border enabled, disabled and unset. If the value of the child is unset it will take the value of the parent.

For 'Styles' I didn't do this since it would involve adding a dummy 'None' style.

Reply:
About the first option, I believe would make more sense to have higher priority in the modules instead of in the outline, so the user set in the outline and override through the modules. In this case, the user would be able to set a default for all modules and override it to a specific module only. And we could also add a small warning below these settings.

I personally prefer this one because we avoid set multiple attributes at one time and the code probably would be much more simple. But if y'all prefer the second option, it's okay for me too.

Reply:
+1 that we should go with the first option with a warning as it is simpler to implement.